src/presentation/api/RootController.ts

```diff
@@ -0,0 +1,68 @@
+import express, { Request, Response } from 'express';
+import { Inject, Service } from 'typedi';
+
+import * as config from '../../config';
+import logger from '../../infrastructure/logging/logger';
+import { ApiError, apiErrorCodes } from '../../api/ApiError';
+import { attachAuthenticationToRequest } from '../../api/middleware/attachAuthenticationToRequest';
+import { wrapAsyncFunction } from '../../api/AsyncRouter';
+
+import accessPass from '../../api/routes/accessPass';
+import { AdminController } from './AdminController';
+import auth from '../../api/routes/auth';
+import countries from '../../api/routes/countries';
+import permissions from '../../api/routes/permissions';
+import roles from '../../api/routes/roles';
+import sharingCode from '../../api/routes/sharingCode';
+import test from '../../api/routes/test';
+import testTypes from '../../api/routes/testTypes';
+import user from '../../api/routes/user';
+
+@Service()
+export class RootController {
+  constructor(private adminController: AdminController) {}
+
+  public routes() {
+    return express()
+      .use('', wrapAsyncFunction(attachAuthenticationToRequest))
+      .use('/v1', accessPass())
+      .use('/v1/admin', this.adminController.routes())
+      .use('/v1', auth())
+      .use('/v1', countries())
+      .use('/v1', permissions())
+      .use('/v1', roles())
+      .use('/v1', sharingCode())
+      .use('/v1', test())
+      .use('/v1', testTypes())
+      .use('/v1/users', user())
```

Comment:
Ah, it's here.  Wonder if we should go the whole hog and switch them all if we're taking this approach?

Reply:
I didn't want to make the PR bigger, but I'll start moving the other stuff in next PRs